### PR TITLE
using `Self` now that it's available in typing_extensions

### PR DESCRIPTION
**Pull request by bandophahita**

### CODE REVIEW

File: screenpy/actions/either.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 from typing import TYPE_CHECKING
 
+from typing_extensions import Self
```
Comment (perrygoy) [inline screenpy/actions/either.py]: Can we put it in `if TYPE_CHECKING`? 🥺

**Review by perrygoy — COMMENTED** [review]
Man look at this, so much cleaner. I only had one small suggestion!

File: screenpy/actions/either.py (hunk 1 above)
Comment (bandophahita) [reply screenpy/actions/either.py]: I sure can.  I even figured out how to get ruff to do that automatically.  See latest commit.

**Review by perrygoy — APPROVED** [review]
Very nice! And what a relief to finally have `Self`.

**Pull request merged**